src/Common/src/System/Net/HttpKnownHeaderNames.TryGetHeaderName.cs

```diff
@@ -226,5 +271,31 @@ private static bool TryMatch(string known, char[] key, int startIndex, int lengt
             name = null;
             return false;
         }
+
+        private unsafe static bool EqualsOrdinal(string left, IntPtr right, int rightLength)
+        {
+            Debug.Assert(left != null);
+            Debug.Assert(right != IntPtr.Zero);
+            Debug.Assert(rightLength > 0);
+
+            // At this point the lengths have already been determined to be equal.
+            Debug.Assert(left.Length == rightLength);
+
+            byte* pRight = (byte*)right;
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                if (left[i] != pRight[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private delegate char CharAt<T>(T key, int index);
```

Comment:
I originally had `Func<...>` but changed them to concrete delegates to make the parameters more clear. Though, it doesn't matter all that much here. I can change it back to `Func<...>`s if you want.